src/FuncSharp/Collections/IEnumerableExtensions_Numeric.cs

```diff
@@ -1,23 +1,51 @@
 ﻿using System.Collections.Generic;
+using System.Diagnostics.Contracts;
 using System.Linq;
 
 namespace FuncSharp
 {
     public static partial class IEnumerableExtensions
     {
+        [Pure]
         public static PositiveInt Sum(this INonEmptyEnumerable<PositiveInt> values)
         {
             return PositiveInt.CreateUnsafe(values.Sum(v => v.Value));
         }
 
+        [Pure]
+        public static PositiveDecimal Sum(this INonEmptyEnumerable<PositiveDecimal> values)
+        {
+            return PositiveDecimal.CreateUnsafe(values.Sum(v => v.Value));
+        }
+
+        public static IOption<PositiveInt> SafeSum(this IEnumerable<PositiveInt> values)
+        {
+            return PositiveInt.Create(values.Sum(v => v.Value));
+        }
+
+        public static IOption<PositiveDecimal> SafeSum(this IEnumerable<PositiveDecimal> values)
+        {
+            return PositiveDecimal.Create(values.Sum(v => v.Value));
+        }
+
         public static NonPositiveInt Sum(this IEnumerable<NonPositiveInt> values)
         {
-            return values.Aggregate(0.AsUnsafeNonPositive(), (a, b) => a + b);
+            return values.Aggregate(NonPositiveInt.Zero, (a, b) => a + b);
+        }
+
+        public static NonPositiveDecimal Sum(this IEnumerable<NonPositiveDecimal> values)
+        {
+            return values.Aggregate(NonPositiveDecimal.Zero, (a, b) => a + b);
```

Comment:
I would much prefer decimal.Add, since .NET Core and higher implements an IEEE 754 compliant representations of floating point unit.

```suggestion
            return values.Aggregate(NonPositiveDecimal.Zero, (a, b) => decimal.Add(a, b));
```

Reply:
Nevermind, that's mostly a float problem